app/src/main/java/to/bitkit/ui/screens/transfer/SpendingConfirmScreen.kt

```diff
@@ -66,11 +73,27 @@ fun SpendingConfirmScreen(
     onLearnMoreClick: () -> Unit = {},
     onAdvancedClick: () -> Unit = {},
     onConfirm: () -> Unit = {},
+    settingsViewModel: SettingsViewModel = hiltViewModel(),
 ) {
+    val context = LocalContext.current
+
     val state by viewModel.spendingUiState.collectAsStateWithLifecycle()
-    val order = state.order ?: return
+
+    val order = state.order ?: run {
+        onCloseClick()
+        return
+    }
     val isAdvanced = state.isAdvanced
 
+    val notificationsGranted by settingsViewModel.notificationsGranted.collectAsStateWithLifecycle()
+
+    RequestNotificationPermissions(
+        onPermissionChange = { granted ->
+            settingsViewModel.setNotificationPreference(granted)
+        },
+        showPermissionDialog = false
+    )
```

Comment:
Should we foresee an optimisation to:
Only show the fg service notification AFTER user approved permissions?!